desktop/cmp/tab/TabSwitcher.js

```diff
@@ -28,51 +34,249 @@ export const [TabSwitcher, tabSwitcher] = hoistCmp.withFactory({
+                    items: [
+                        icon,
+                        span(title),
+                        button({
+                            omit: !showRemoveAction,
+                            tabIndex: -1,
+                            icon: Icon.x(),
+                            onClick: () => tab.containerModel.removeTab(tab)
+                        })
+                    ]
+                })
             });
         });
 
-        return blueprintTabs({
-            id,
-            vertical,
-            onChange: (tabId) => model.activateTab(tabId),
-            selectedTabId: activeTabId,
-            items,
-            ...props,
-            animate: withDefault(animate, false),
-            className: classNames(className, `xh-tab-switcher--${orientation}`)
+        return div({
+            className: classNames(
+                className,
+                `xh-tab-switcher--${orientation}`,
+                enableOverflow ? `xh-tab-switcher--overflow-enabled` : null
+            ),
+            items: [
+                div({
+                    ref,
+                    className: `xh-tab-switcher__scroll`,
+                    item: blueprintTabs({
+                        id,
+                        vertical,
+                        animate,
+                        items,
+                        selectedTabId: activeTabId,
+                        onChange: (tabId) => model.activateTab(tabId)
+                    })
+                }),
+                overflowMenu({
+                    tabs: impl.overflowTabs,
+                    vertical
+                })
+            ]
         });
     }
 });
 
 TabSwitcher.propTypes = {
-    /** True to animate the indicator when switching tabs. False (default) to change instantly. */
-    animate: PT.bool,
-
     /** Primary component model instance. */
     model: PT.instanceOf(TabContainerModel),
 
     /** Relative position within the parent TabContainer. Defaults to 'top'. */
-    orientation: PT.oneOf(['top', 'bottom', 'left', 'right'])
-};
\ No newline at end of file
+    orientation: PT.oneOf(['top', 'bottom', 'left', 'right']),
+
+    /** True to animate the indicator when switching tabs. False (default) to change instantly. */
+    animate: PT.bool,
+
+    /** Enable scrolling and place tabs that overflow into a menu. Default to false. */
+    enableOverflow: PT.bool,
+
+    /** Size (in px) to render tabs */
+    tabSize: PT.number,
+
+    /** Minimum size (in px) to render tabs */
+    tabMinSize: PT.number,
+
+    /** Maximum size (in px) to render tabs */
+    tabMaxSize: PT.number
+};
+
+//-----------------
+// Implementation
+//-----------------
+const overflowMenu = hoistCmp.factory({
+    render({model, tabs, vertical}) {
+        if (isEmpty(tabs)) return null;
+
+        const items = tabs.map(tab => {
+            const {id, title: text, icon, disabled} = tab;
+            return menuItem({
+                icon,
```

Comment:
+ Should this menu get a width of tabSize/maxTabSize/minTabSize for consistency?  Just thinking about very long tab names blowing this menu up, and getting a consistent.
+ Should these menu items get the tab tooltops?